src/main/java/zmq/io/net/tcp/TcpAddress.java

```diff
@@ -114,24 +123,28 @@ public InetSocketAddress resolve(String name, boolean ipv6, boolean local)
 
         InetAddress addrNet = null;
 
+        // '*' as unspecified address is not accepted in Java
+        // '::' for IPv6 is accepted
         if (addrStr.equals("*")) {
-            addrStr = "0.0.0.0";
+            addrStr = ipv6 ? "::" : "0.0.0.0";
         }
+
         try {
-            for (InetAddress ia : InetAddress.getAllByName(addrStr)) {
-                if (ipv6 && !(ia instanceof Inet6Address)) {
-                    continue;
-                }
-                addrNet = ia;
-                break;
+            InetAddress[] addresses = InetAddress.getAllByName(addrStr);
+            if (ipv6) {
+                // prefer IPv6: return the first ipv6 or the first value if not found
+                addrNet = Arrays.stream(InetAddress.getAllByName(addrStr)).filter(i -> i instanceof Inet6Address).findFirst().orElseGet(() -> addresses[0]);
```

Comment:
Isn't `InetAddress.getAllByName(addrStr)` equivalent to `addresses` here?

Reply:
Indeed